.github/workflows/ci.yaml

```diff
@@ -43,28 +44,36 @@ jobs:
 
   release:
     name: Release
+    permissions:
+      contents: write
+      packages: write
+      issues: write
     needs: [test, build]
     if: github.ref == 'refs/heads/main'
     runs-on: ubuntu-latest
+
     steps:
-    - name: Checkout
-      uses: actions/checkout@692973e3d937129bcbf40652eb9f2f61becf3332 # v4.1.7
-      with:
-        fetch-depth: 0
-        persist-credentials: false
+      - name: Checkout
+        uses: actions/checkout@692973e3d937129bcbf40652eb9f2f61becf3332 # v4.1.7
+        with:
+          fetch-depth: 0
+
+      - name: Setup Node
+        uses: actions/setup-node@60edb5dd545a775178f52524783378180af0d1f8 # v4.0.2
+        with:
+          # See https://github.com/semantic-release/semantic-release/blob/master/docs/support/node-version.md
+          node-version: "20.14.0"
+
+      - name: Fetch build
+        uses: actions/download-artifact@65a9edc5881444af0b9093a5e628f2fe47ea3b2e # v4.1.7
+        with:
+          name: artifact
+          path: bin
```

Comment:
Sorry I'm being a bit dim, at which point in this pipeline does the Go code get compiled and packaged into an executable?

Reply:
that happens on [line 39](https://github.com/guardian/bucket-blocker/blob/a3d7686e8c0b2889f867f63481ce27333c4dda4a/.github/workflows/ci.yaml#L39) - was done in a previous PR!